store/rootmulti/store.go

```diff
@@ -192,7 +192,20 @@ func (rs *Store) loadVersion(ver int64, upgrades *types.StoreUpgrades) error {
 	// load each Store (note this doesn't panic on unmounted keys now)
 	var newStores = make(map[types.StoreKey]types.CommitKVStore)
 
-	for key, storeParams := range rs.storesParams {
+	storesKeys := make([]types.StoreKey, 0, len(rs.storesParams))
+
+	for key := range rs.storesParams {
+		storesKeys = append(storesKeys, key)
+	}
+	if upgrades != nil {
+		// deterministic iteration order for upgrades
+		sort.Slice(storesKeys, func(i, j int) bool {
+			return storesKeys[i].Name() < storesKeys[j].Name()
+		})
+	}
```

Comment:
so, we are are sorting the keys only when doing an upgrade. It seams that we are safe to merge it into 0.44 - because it won't invoke an update for chains which are already running on 0.44. Same for `RunMigrations` method [below](https://github.com/cosmos/cosmos-sdk/pull/10189/files#diff-98d29a1c94b0467f2a1276f5b8e28ed6c2e74385d4659cc6f675814242dc0122R397).
What do you think @amaurym ?

Reply:
Hmm I'm not sure. Imagine an existing chain is running 0.44.0. Then, a full node wants to sync, uses 0.44.1 (with this patch): won't there be a hash mismatch right after the upgrade?

Reply:
if a chain is using 0.44 and updates the binary to 0.44.1 and then syncs, then there is no migration and no store is added/renamed/deleted. So the sorting code won't run.

Reply:
> Hmm I'm not sure. Imagine an existing chain is running 0.44.0. Then, a full node wants to sync, uses 0.44.1 (with this patch): won't there be a hash mismatch right after the upgrade?

that would likely only apply if that chain ran upgrade 0.42->0.44 (where they could encounter this issue anyway)... so far, I only of BandChain and that was upgraded via a state export (instead of using the upgrade module)

Reply:
Sounds good, let's backport to 0.44.1 then!